src/OctoshiftCLI.Tests/BbsClientTests.cs

```diff
@@ -115,6 +115,70 @@ public async Task GetAsync_Logs_The_Response_Status_Code_And_Content()
         _mockOctoLogger.Verify(m => m.LogVerbose($"RESPONSE ({_httpResponse.StatusCode}): {EXPECTED_RESPONSE_CONTENT}"), Times.Once);
     }
 
+    [Fact]
+    public async Task GetAsync_Retries_On_ServiceUnavailable()
+    {
+        // Arrange
+        using var firstHttpResponse = new HttpResponseMessage(HttpStatusCode.ServiceUnavailable)
+        {
+            Content = new StringContent("FIRST_RESPONSE")
+        };
+        using var secondHttpResponse = new HttpResponseMessage(HttpStatusCode.OK)
+        {
+            Content = new StringContent("SECOND_RESPONSE")
+        };
+        var handlerMock = new Mock<HttpMessageHandler>();
+        handlerMock
+            .Protected()
+            .SetupSequence<Task<HttpResponseMessage>>(
+                "SendAsync",
+                ItExpr.Is<HttpRequestMessage>(req => req.Method == HttpMethod.Get),
+                ItExpr.IsAny<CancellationToken>())
+            .ReturnsAsync(firstHttpResponse)
+            .ReturnsAsync(secondHttpResponse);
+
+        using var httpClient = new HttpClient(handlerMock.Object);
+        var bbsClient = new BbsClient(_mockOctoLogger.Object, httpClient, null, _retryPolicy, USERNAME, PASSWORD);
+
+        // Act
+        var returnedContent = await bbsClient.GetAsync(URL);
+
+        // Assert
+        returnedContent.Should().Be("SECOND_RESPONSE");
+    }
+
+    [Fact]
+    public async Task GetAsync_Retries_On_Unauthorized()
+    {
+        // Arrange
+        using var firstHttpResponse = new HttpResponseMessage(HttpStatusCode.Unauthorized)
+        {
+            Content = new StringContent("FIRST_RESPONSE")
+        };
+        using var secondHttpResponse = new HttpResponseMessage(HttpStatusCode.OK)
+        {
+            Content = new StringContent("SECOND_RESPONSE")
+        };
+        var handlerMock = new Mock<HttpMessageHandler>();
+        handlerMock
+            .Protected()
+            .SetupSequence<Task<HttpResponseMessage>>(
+                "SendAsync",
+                ItExpr.Is<HttpRequestMessage>(req => req.Method == HttpMethod.Get),
+                ItExpr.IsAny<CancellationToken>())
+            .ReturnsAsync(firstHttpResponse)
+            .ReturnsAsync(secondHttpResponse);
+
+        using var httpClient = new HttpClient(handlerMock.Object);
+        var bbsClient = new BbsClient(_mockOctoLogger.Object, httpClient, null, _retryPolicy, USERNAME, PASSWORD);
+
+        // Act
+        var returnedContent = await bbsClient.GetAsync(URL);
+
+        // Assert
+        returnedContent.Should().Be("SECOND_RESPONSE");
+    }
```

Comment:
Use `[Theory]` and merge the two tests.

Reply:
Added in 7b0d6fc6f9059a2270d480c94d1d42e068d9fc0b :smile: